pkg/auth/tokenparser_test.go

```diff
@@ -273,7 +273,27 @@ func (s *TestTokenParserSuite) TestTokenParser() {
 		// validate token
 		_, err = tokenParser.FromString(jwt0string)
 		require.Error(s.T(), err)
-		require.EqualError(s.T(), err, "token is not valid yet")
+		require.EqualError(s.T(), err, "token has invalid claims: token is not valid yet")
+	})
+
+	s.Run("signature is good and token expiration is within leeway", func() {
```

Comment:
This is a new test to make sure the leeway window is taken into account when we parse the token. It tests that we migrated the leeway thing correctly.